fendermint/vm/topdown/src/sync/pointers.rs

```diff
@@ -31,6 +31,11 @@ impl SyncPointers {
     pub fn set_tail(&mut self, height: BlockHeight, hash: BlockHash) {
         self.tail = Some((height, hash));
     }
+
+    pub fn reset(&mut self, head: BlockHeight) {
```

Comment:
Here you are just resetting the pointers. Is there no need to clean the content from the cache? We may be leaking outdated information by just changing the pointers, right?